### PR TITLE
SAM-3066 remove unnecessary indexes

### PR DESCRIPTION
**Pull request by ottenhoff**
Hibernate will create an index on all constraints. So there is no need to create an additional index.

Redundant indexes only slow things down (and make the database bigger).

### CODE REVIEW

**Comment by ottenhoff** [comment]
Any comments?

File: reference/docs/conversion/sakai_12_mysql_conversion.sql [hunk 1]
```diff
@@ -208,3 +208,15 @@ ALTER TABLE pasystem_banner_dismissed DROP COLUMN user_eid;
...
 alter table GB_GRADABLE_OBJECT_T alter column IS_EXTRA_CREDIT drop default;
 alter table GB_GRADABLE_OBJECT_T alter column HIDE_IN_ALL_GRADES_TABLE drop default;
+
+-- BEGIN SAM-3066 remove unecessary indexes because Hibernate always create an index on an FK
+ALTER TABLE SAM_PUBLISHEDITEM_T DROP INDEX SAM_PUBITEM_SECTION_I;
+ALTER TABLE SAM_PUBLISHEDITEMFEEDBACK_T DROP INDEX SAM_PUBITEMFB_ITEM_I;
+ALTER TABLE SAM_PUBLISHEDITEMMETADATA_T DROP INDEX SAM_PUBITEMMETA_ITEM_I;
+ALTER TABLE SAM_PUBLISHEDITEMTEXT_T DROP INDEX SAM_PUBITEMTEXT_ITEM_I;
+ALTER TABLE SAM_PUBLISHEDSECTION_T DROP INDEX SAM_PUBSECTION_ASSESSMENT_I;
+ALTER TABLE SAM_PUBSECTION_ASSESSMENT_I DROP INDEX SAM_PUBITEM_SECTION_I;
```
Comment (danielmerino) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: There is a typo here, the table has an index name and it fails. Looking other commits, I think it should be:
ALTER TABLE SAM_PUBLISHEDITEM_T DROP INDEX SAM_PUBITEM_SECTION_I;
but I don't have this index in my table SAM_PUBLISHEDITEM_T in 10.5 so I'm not sure at all.

Comment (ottenhoff) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: If you don't have an index on that column, ignore the error and continue running.

Comment (danielmerino) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: Yeah, but SAM_PUBSECTION_ASSESSMENT_I is not a table, right? It seems to be a typo here.

Comment (ottenhoff) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: SAM_PUBSECTION_ASSESSMENT_I is the index name not the table name. I don't understand what you are reporting here.

Comment (danielmerino) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: In line 217 the index is correctly used, but in line 218 is used again and it's incorrect, isn't it?

Comment (ottenhoff) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: Yes, you are correct. I will create a PR.

Comment (danielmerino) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: Thanks, Sam!

**Pull request merged**